code/implementation/host-stats/src/main/java/io/cattle/platform/host/stats/api/HostStatsLinkHandler.java

```diff
@@ -23,56 +28,79 @@
 
-    private static final DynamicStringProperty HOST_STATS_PATH = ArchaiusUtil.getString("host.stats.path");
+    private static final DynamicStringProperty HOST_STATS_PATH = ArchaiusUtil.getString("link.hoststats.path");
+
+    private static final String PROJECT_PATH = "/project/";
 
     HostApiService hostApiService;
     ObjectManager objectManager;
+    TokenService tokenService;
 
     @Override
     public String[] getTypes() {
-        return new String[] { InstanceConstants.TYPE_CONTAINER, HostConstants.TYPE };
+        return new String[] { HostConstants.TYPE, ProjectConstants.TYPE };
     }
 
     @Override
     public boolean handles(String type, String id, String link, ApiRequest request) {
-        return HostStatsConstants.LINK_STATS.equals(link);
+        return StatsConstants.HOST_STATS.equals(link) && ("project".equals(type) || HostConstants.TYPE.equals(type));
     }
 
     @Override
     public Object link(String name, Object obj, ApiRequest request) throws IOException {
-        Host host = null;
-        Instance instance = null;
-
-        if (obj instanceof Instance) {
-            instance = (Instance) obj;
-            host = DockerUtils.getHostFromContainer(objectManager, instance);
-        } else if (obj instanceof Host) {
-            host = (Host) obj;
+        List<Host> hosts = new ArrayList<>();
+        boolean project = false;
+        if (obj instanceof Host) {
+            Host origHost = (Host) obj;
+            hosts.add(origHost);
         }
 
-        if (host == null) {
-            return null;
+        if (obj instanceof Account && AccountConstants.PROJECT_KIND.equals(((Account) obj).getKind())) {
+            hosts.addAll(objectManager.mappedChildren(obj, Host.class));
+            project = true;
         }
 
-        HostApiAccess apiAccess = hostApiService.getAccess(host.getId(), Collections.<String, Object> emptyMap());
-        if (apiAccess == null) {
+        if (hosts.size() == 0) {
             return null;
         }
 
-        StringBuilder url = new StringBuilder(HostApiUtils.HOST_API_PROXY_SCHEME.get());
-        url.append("://").append(apiAccess.getHostAndPort());
-        url.append(HOST_STATS_PATH.get());
-
-        if (instance != null) {
-            String dockerId = DockerUtils.getDockerIdentifier(instance);
-            url.append("/").append(dockerId);
+        String metaUrl = null;
+        StatsAccess meta = new StatsAccess();
+        List<StatsAccess> serviceStatsQuery = new ArrayList<>();
+        for (Host host : hosts) {
+            if (!"active".equals(host.getState())) {
+                continue;
+            }
+            Map<String, Object> payload = new HashMap<>();
+            payload.put("resourceId", ApiContext.getContext().getIdFormatter().formatId(objectManager.getType(host), host.getId()));
+            HostApiAccess apiAccess = hostApiService.getAccess(host.getId(), payload);
+            if (apiAccess == null) {
+                return null;
```

Comment:
I was thinking if just 1 host out of 10 gets a null apiAccess, we could just skip it and return the other 9.